<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>manejo de errores</title>

    <script>
        function hayONoHay() {
            return new Promise((resolve, reject) => {
                const val = Math.round(Math.random() * 1); // 0 o 1, generaciÃ³n de random

                val ? resolve('Suertudo!!') : reject('NO! ðŸ˜ ');
            });
        }

        async function msg() {
            try {
                const msg = await hayONoHay();
                console.log(msg);
            } catch (err) {
                console.log(err);
            }
        } 

        msg();
        msg();
        msg();
        msg();
        msg();
        msg();
        msg();
        msg();
        msg();
        msg();
        msg();


        
    </script>
</head>

<body>

</body>

</html>